packages/monitoring/src/adapters/adapter.ts

```diff
@@ -67,8 +71,9 @@ export interface Transaction {
   /**
    * Start a measurement.
    * @param name The name of the measurement. This name will be treated as the identifier of the measurement.
+   * @param data Option data to be send at the start of measurements.
    */
-  startMeasurement: (name: string) => void;
+  startMeasurement: (name: string, data?: { [key: string]: number | string }) => void;
```

Comment:
Replace the type of the `data` object with : `Record<string, number | string>`

Also, could you modify the type definitions in the whole monitoring project and every time there is something like `{ [key: X]: Y}`, can you replace it with `Record<X, Y>` please?

Reply:
yeah sure